### PR TITLE
feat: Health signal exposure (HealthSnapshot, tracker, Trading facade)

### PR DESCRIPTION
**Pull request by blahmonkey**
**NOTE:** The author must always **first review the PR themselves**, before requesting review from others, that means carefully having reviewed the full diff in GitHub.

## What does this PR change?

Implements MMVP-TD issue [#68](https://github.com/Vaquum/Praxis/issues/68) Health subitems (Health.1–Health.4) — the Trading sub-system's contribution to RFC-4001 §Health. After this PR, Manager (Nexus) can pull a `HealthSnapshot` per account so its `HealthEvaluator` can drive `ACTIVE` / `REDUCE_ONLY` / `HALTED` mode transitions.

**New files:**
- [`praxis/core/domain/health_snapshot.py`](praxis/core/domain/health_snapshot.py) — frozen `HealthSnapshot` dataclass with finite/non-negative invariants, `[0, 1]` ratio bounds for `failure_rate` / `rate_limit_headroom`, and int-only `consecutive_failures`
- [`praxis/core/health_tracker.py`](praxis/core/health_tracker.py) — `HealthTracker` rolling-window collector; thread-safe `record_request(latency_ms, succeeded)`; on-demand p99 / failure-rate / `consecutive_failures` composition
- [`docs/Health.md`](docs/Health.md) — page describing the snapshot contract, collection path, and cross-thread call pattern
- [`tests/test_health_snapshot.py`](tests/test_health_snapshot.py) — 21 contract tests
- [`tests/test_health_tracker.py`](tests/test_health_tracker.py) — 10 tracker tests

**Modified files:**
- [`praxis/infrastructure/binance_adapter.py`](praxis/infrastructure/binance_adapter.py) — owns one `HealthTracker` per registered account; `_request_with_retry` records latency + outcome once per logical request (retries internal, not double-counted); new `rate_limit_utilization` and `clock_drift_ms` properties; `sync_clock_drift()` populates drift from `/api/v3/time`; `get_health_snapshot(account_id)` composes a snapshot. `_health_lock` guards registry mutation, lookup, snapshot composition, and `_clock_drift_ms` assignment.
- [`praxis/infrastructure/venue_adapter.py`](praxis/infrastructure/venue_adapter.py) — `get_health_snapshot(account_id) -> HealthSnapshot` added to the `VenueAdapter` Protocol so the Trading facade can delegate without depending on the concrete adapter.
- [`praxis/trading.py`](praxis/trading.py) — `Trading.get_health_snapshot(account_id)` async method (start-required, intentionally not `account_ready`-required) so Manager can call via `asyncio.run_coroutine_threadsafe(trading.get_health_snapshot(...), trading.loop)` from its own thread.
- [`tests/test_binance_adapter.py`](tests/test_binance_adapter.py) — `TestHealthSignals` class with 12 tests covering REST recording, rate-limit utilisation, snapshot composition, per-account isolation, and `sync_clock_drift` happy/non-OK/transport-error branches.
- [`tests/test_trading.py`](tests/test_trading.py) — 2 tests for facade delegation and start-required behaviour.
- [`tests/test_venue_adapter.py`](tests/test_venue_adapter.py) — Protocol conformance test extended.
- [`docs/TechnicalDebt.md`](docs/TechnicalDebt.md) — adds TD-018 noting that `sync_clock_drift`'s midpoint estimator ignores asymmetric network latency.
- `CHANGELOG.md` — v0.40.0 entry
- `pyproject.toml` — version bump 0.39.0 → 0.40.0

**Design decisions:**
- **Pull, not push.** Manager polls; outcome-queue health events are out of scope for now (issue #68 leaves it as either/or; pull is the simpler contract).
- **`async def Trading.get_health_snapshot` with sync body.** The signature exists so a sync Manager thread can dispatch via `run_coroutine_threadsafe` without blocking the loop — composition itself is in-memory.
- **`rate_limit_headroom` carries utilisation semantics** (higher is worse). Field name preserved for parity with the Manager-side `HealthSnapshot` in Nexus; intentional, documented in `docs/Health.md`.
- **Per-account `HealthTracker`** for latency + failures (account-scoped); venue-wide for `rate_limit_utilization` and `clock_drift_ms` (one venue per adapter).
- **`_health_lock`** added pre-emptively against the same race pattern Copilot flagged on `_outcome_queues` in PR #69 (Launcher mutates registry from a non-loop thread; loop reads on the hot path).
- **Retries don't double-count.** `_request_with_retry` records latency from the entry timestamp through to the resolved outcome — one logical row per call.

## Checklist

- [x] I have reviewed full diff in "Files changed"
- [x] I left no unnecessary files in the changes
- [x] I ran `python tests/run.py` locally without errors (where applicable)
- [x] I updated `/docs` (if behavior/API/config/user/etc changed)
- [x] I added and/or updated docstrings as per [Writing Docstrings](https://github.com/Vaquum/dev-docs/blob/main/src/Writing-Docstrings.md) (for any changed public functions/classes)
- [x] I updated CHANGELOG.md (unless only docs or other non-code aspect was changed)
- [x] I updated pyproject.toml (unless only docs or other non-code aspect was changed)
- [x] I added and/or updated tests (if behavior changed or new code paths added)
- [x] I validated changes manually
- [x] I validated changes with LLM
- [x] I removed any extraneous examples/comments
- [ ] I linked issue to auto-close on merge (e.g., "Fixes #123") when applicable

Refs #68 (Health.1–Health.4)

### CODE REVIEW

**Review by zero-bang — DISMISSED** [review]
Clean, well-structured implementation of Health.1–Health.4. Reviewed the full diff, traced all error paths, locking boundaries, and Protocol conformance. Key observations:

**Correctness**: All `_record_health` calls in `_request_with_retry` fire exactly once per logical request — verified every exit path (success return, TransientError break, RateLimitError raise, VenueError raise, transport-error break, post-loop exhaustion). No double-counting.

**Thread-safety**: `_health_lock` correctly guards the tracker registry and `_clock_drift_ms` writes. Individual tracker operations use their own internal lock — good separation. `rate_limit_utilization` reading `_used_weight`/`_weight_limit` outside a lock is acceptable: both are only written from the asyncio loop thread, and for a health signal the potential for a slightly stale pair is harmless.

**Design**: Per-account tracker + venue-wide metrics composed at snapshot time is the right split. `Trading.get_health_snapshot` being start-required but not account-ready-required is a good call for observability. The sync Protocol method with async Trading wrapper for cross-thread dispatch is clean.

**Tests**: 46 new/modified tests cover invariants, edge cases (NaN, bool, negative, boundary ratios), rolling window truncation, per-account isolation, clock drift happy/error paths, and facade delegation. Solid coverage.

**Docs**: `Health.md` code example matches the actual API. TD-018 is an honest acknowledgment of the midpoint drift limitation with a clear "when to fix" gate.

No blocking issues found. LGTM.

File: praxis/infrastructure/binance_adapter.py [hunk 1]
```diff
@@ -1380,6 +1398,112 @@ async def query_order_book(
...
+    @property
+    def clock_drift_ms(self) -> float:
+
+        '''
+        Last measured absolute clock drift from the exchange in milliseconds.
+
+        Returns 0.0 until sync_clock_drift has been called successfully.
+        '''
+
+        return self._clock_drift_ms
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply praxis/infrastructure/binance_adapter.py]: Addressed in 76c5743. `clock_drift_ms` getter now also takes `_health_lock`, matching the lock taken by `sync_clock_drift` on the writer side.

File: praxis/trading.py [hunk 2]
```diff
@@ -748,6 +749,35 @@ def submit_abort(self, abort: TradeAbort) -> None:
         self._require_account_ready(abort.account_id)
         self._inbound.submit_abort(abort)
 
+    async def get_health_snapshot(self, account_id: str) -> HealthSnapshot:
+        '''Return a HealthSnapshot for an account.
+
+        Body is synchronous (snapshot composition is in-memory). The method
+        is declared `async` so a Manager thread can dispatch the call
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply praxis/trading.py:756]: No code change. RUF029 is currently a preview rule and is not enabled in this repo (verified: `ruff check --select RUF029` warns "Selection RUF029 has no effect because preview is not enabled"). I tried adding `# noqa: RUF029` with a rationale and ruff RUF100 immediately rejected it as an unused noqa for a non-enabled rule, so the noqa cannot live there today. The design intent (sync body inside an async signature so a Manager thread can dispatch via `run_coroutine_threadsafe` without blocking the loop) is already documented in the docstring. Happy to revisit if RUF029 is later promoted out of preview.

**Review by zero-bang — APPROVED** [review]
